ENH: Add utility function to get ROI box as vtkPolyData

lassoan opened the pull request:
Getting the ROI box as polydata is useful for example for creating a model or segment from an ROI node.

Sunderlandkyl reviewed (approved):
LGTM!

jcfr commented on Modules/Loadable/Markups/MRML/vtkMRMLMarkupsROINode.h:
```diff
@@ -240,6 +240,11 @@ class  VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsROINode : public vtkM
 
+  /// Create ROI box as surface mesh in the world coordinate system as a new vtkPolyData object.
+  /// In C++, the owner must take ownership of the returned object (delete when no longer used).
+  /// In Python, the returned object automatically takes ownership of the returned object (no need to manually delete).
```
To be specific the ownership is now correctly transferred to python, so conceptually Python is responsible for "deleting" the object.

jcfr replied:
```suggestion
  /// In Python, specifying VTK_NEWINSTANCE ensures the ownership is transferred to Python where garbage collection takes care of re-referencing the object.
```

jcfr commented on Modules/Loadable/Markups/MRML/vtkMRMLMarkupsROINode.h:
```diff
@@ -240,6 +240,11 @@ class  VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsROINode : public vtkM
   vtkGetObjectMacro(ImplicitFunctionWorld, vtkImplicitFunction);
 
+  /// Create ROI box as surface mesh in the world coordinate system as a new vtkPolyData object.
+  /// In C++, the owner must take ownership of the returned object (delete when no longer used).
```
```suggestion
  /// In C++, the owner must take ownership of the returned object (delete when no longer referenced).
```

lassoan commented on Modules/Loadable/Markups/MRML/vtkMRMLMarkupsROINode.h:
```diff
@@ -240,6 +240,11 @@ class  VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsROINode : public vtkM
 
+  /// Create ROI box as surface mesh in the world coordinate system as a new vtkPolyData object.
+  /// In C++, the caller is expected to take ownership of the returned object (delete when no longer referenced).
+  /// In Python, specifying VTK_NEWINSTANCE ensures the ownership is transferred to Python where garbage collection takes care of re-referencing the object.
```
```suggestion
  /// In Python, specifying VTK_NEWINSTANCE ensures the ownership is transferred to Python where garbage collection takes care of de-referencing the object.
```

lassoan commented:
@jcfr I've incorporated your comments and moved the explanation to the developer documentation (it was a bit too verbose to repeat at every places where `VTK_NEWINSTANCE` is used).
